assets/account/src/lib.rs

```diff
@@ -6,30 +6,18 @@ blueprint! {
     }
 
     impl Account {
-        pub fn new(public_key: EcdsaPublicKey) -> ComponentId {
-            let key = NonFungibleId::new(public_key.to_vec());
-            let auth_address = NonFungibleAddress::new(ECDSA_TOKEN, key);
-
+        pub fn new(withdraw_auth: ProofRule) -> ComponentId {
```

Comment:
Minor: not sure if `ProofRule` is the right name, as it exposes the implementation details. Maybe `AccessRule`?